pkg/controller/ingress-shim/sync.go

```diff
@@ -204,6 +218,35 @@ func (c *Controller) buildCertificates(ing *extv1beta1.Ingress, issuer v1alpha1.
 	return newCrts, updateCrts, nil
 }
 
+func (c *Controller) findUnrequiredCertificates(ing *extv1beta1.Ingress) ([]*v1alpha1.Certificate, error) {
+	var unrequired []*v1alpha1.Certificate
+	crts, err := c.certificateLister.Certificates(ing.Namespace).List(labels.Everything())
+	if err != nil {
+		return nil, err
+	}
+
+	for _, crt := range crts {
+		if isUnrequiredCertificate(crt, ing) {
+			unrequired = append(unrequired, crt)
+		}
+	}
+
+	return unrequired, nil
+}
+
+func isUnrequiredCertificate(crt *v1alpha1.Certificate, ing *extv1beta1.Ingress) bool {
+	if !metav1.IsControlledBy(crt, ing) {
```

Comment:
I think it might be possible for us to be cleverer with the label selector in `findUnrequiredCertificates` in order fo filter the list of certificate's we consider to only those that *are* controlled/owned by the given Ingress resource. I'm not too certain though, but #sig-api-machinery on the k8s slack will be able to help.

It may not be important to do now, but it could be interesting for you to look into. If you don't want to do it now, could you add a TODO here so we don't forget to consider this option in future? 😄

Reply:
Certificates take all the same labels as their owning Ingress, and I imagine an Ingress doesn't add a label onto itself identifying itself so I think we would want to add another label onto new Certificates which links it to a certain Ingress in order to use a more sophisticated selector  - maybe the Ingress' UID?

I will add a TODO for now.